data_safe_haven/logging/logger.py

```diff
@@ -20,7 +21,9 @@ def get_console_handler() -> RichHandler:
     return next(h for h in get_logger().handlers if isinstance(h, RichHandler))
 
 
-def get_logger() -> logging.Logger:
+def get_logger(*, disable_logging: bool = False) -> logging.Logger:
+    if disable_logging:
+        return NonLoggingSingleton()
     return logging.getLogger("data_safe_haven")
```

Comment:
I wonder if

```suggestion
def get_logger() -> logging.Logger:
    return logging.getLogger("data_safe_haven")
    
    
def get_null_logger() -> logging.Logger:
    return NonLoggingSingleton()
```

Would be clearer. Or changing the argument from `disable_logging` to `log_to_null`?
Saying, give me a logger and disable logging just sounds confusing to me.

Reply:
Having things like `sdk = AzureSDK(disable_logging=True)` is good though.

Reply:
Done in 1aa489063